src/yarpWholeBodyStates.cpp

```diff
@@ -1175,34 +1234,54 @@ bool yarpWholeBodyEstimator::computeWorldRootRotoTranslation(double *q_temp)
       world_H_rootLink = world_H_reference*referenceLink_H_rootLink ;
 
       int ctr;
-
       for (ctr=0;ctr<3;ctr++)
       {
-	estimates.lastBasePos(ctr) = world_H_rootLink.p[ctr];
+        estimates.lastBasePos(ctr) = world_H_rootLink.p[ctr];
       }
       for (ctr=0;ctr<9;ctr++)
       {
-	estimates.lastBasePos(3+ctr) = world_H_rootLink.R.data[ctr];
+        estimates.lastBasePos(3+ctr) = world_H_rootLink.R.data[ctr];
       }
       return(true);
   }
   else
-
-    return(false);
+        return(false);
 }
-
-bool yarpWholeBodyEstimator::setWorldBaseLinkName(std::string linkName)
+bool yarpWholeBodyEstimator::computeBaseVelocity(double* qj,double* dqj)
 {
+
   if(wholeBodyModel!=NULL)
   {
-    yInfo()<<"Reference link set as world was "<<robot_reference_frame_link;
-    wholeBodyModel->getFrameList().idToIndex(linkName.c_str(),robot_reference_frame_link);
-    yInfo()<<", now it is set to "<<robot_reference_frame_link<<"\n\n\n";
+    int dof = estimates.lastQ.size();
+//     new (&dqjVect) Eigen::Map<Eigen::VectorXd>(dqj,dof);
+
+//     Eigen::Map<Eigen::VectorXd> dqjVect(dqj,dof);
+//     Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> complete_jacobian(6,dof+6);
+//     Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> joint_jacobian(6,dof);
+//     Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> tempMatForComputation(6,dof);
+//     Eigen::VectorXd dvbVect(6);
+//     new (&rotationalVelocityWrapper) Eigen::Map<Eigen::VectorXd>(estimates.lastBaseVel.data(), estimates.lastBaseVel.size());
+
+    dvbVect.setZero();
+    complete_jacobian.setZero();
+    joint_jacobian.setZero();
+    floatingBase_jacobian.setZero();
+    tempMatForComputation.setZero();
+
+    wholeBodyModel->computeJacobian(qj,world_H_rootLink,robot_reference_frame_link,complete_jacobian.data());
+    floatingBase_jacobian = complete_jacobian.leftCols(6);
+    joint_jacobian = complete_jacobian.rightCols(dof);
+
+    tempMatForComputation = (floatingBase_jacobian.inverse()*joint_jacobian);
+    tempMatForComputation*=-1.0;
+
+    dvbVect =tempMatForComputation*dqjVect;
```

Comment:
Can we do this operation with only one shot? Something like 

dvbVect =-floatingBase_jacobian.inverse()*joint_jacobian*dqjVect